usaspending_api/disaster/v2/views/elasticsearch_base.py

```diff
@@ -0,0 +1,136 @@
+from usaspending_api.disaster.v2.views.disaster_base import DisasterBase
+from usaspending_api.search.v2.elasticsearch_helper import (
+    get_scaled_sum_aggregations,
+    get_number_of_unique_terms_for_awards,
+)
+
+
+class ElasticsearchDisasterBase(DisasterBase):
+
+    query_fields: List[str]
+    agg_key: str
+    is_loans: bool = False
+
+    filter_query: ES_Q
+    bucket_count: int
+
+    pagination: Pagination  # Overwritten by a pagination mixin
+    sort_column_mapping: Dict[str, str]  # Overwritten by a pagination mixin
+    sum_column_mapping: Dict[str, str]  # Overwritten by a pagination mixin
+
+    @cache_response()
+    def post(self, request: Request) -> Response:
+        # Handle "query" filter outside of QueryWithFilters
+        query = self.filters.pop("query", None)
+
+        # If for a loans endpoint then filter on Loans
+        filters = self.filters
+        if self.is_loans:
+            filters.update({"award_type_codes": list(loan_type_mapping)})
+        self.filter_query = QueryWithFilters.generate_awards_elasticsearch_query(filters)
+
+        if query:
+            self.filter_query.must.append(
+                ES_Q("query_string", query=query, default_operator="OR", fields=self.query_fields)
+            )
+
+        # Ensure that only non-zero values are taken into consideration
+        non_zero_queries = []
+        for field in self.sum_column_mapping.values():
+            non_zero_queries.append(ES_Q("range", **{field: {"gt": 0}}))
+            non_zero_queries.append(ES_Q("range", **{field: {"lt": 0}}))
+        self.filter_query.must.append(ES_Q("bool", should=non_zero_queries, minimum_should_match=1))
+
+        self.bucket_count = get_number_of_unique_terms_for_awards(self.filter_query, f"{self.agg_key}.hash")
+
+        results = self.query_elasticsearch()
+
+        return Response(
+            {
+                "results": results[: self.pagination.limit],
+                "page_metadata": get_pagination_metadata(
+                    self.bucket_count, self.pagination.limit, self.pagination.page
+                ),
+            }
+        )
+
+    @abstractmethod
+    def build_elasticsearch_result(self, response: dict) -> List[dict]:
+        pass
+
+    def build_elasticsearch_search_with_aggregations(self) -> Optional[AwardSearch]:
+        """
+        Using the provided ES_Q object creates an AwardSearch object with the necessary applied aggregations.
+        """
+        # Create the initial search using filters
+        search = AwardSearch().filter(self.filter_query)
+
+        pagination_values = {
+            "from": (self.pagination.page - 1) * self.pagination.limit,
+            "size": self.pagination.limit + 1,
+        }
+        sort_values = {"order": {self.sort_column_mapping[self.pagination.sort_key]: self.pagination.sort_order}}
+
+        if self.agg_key == settings.ES_ROUTING_FIELD:
+            size = self.pagination.upper_limit
+            shard_size = size
```

Comment:
Unclear what's happening here exactly. Could use a comment. 

Is this special handling for when we're aggregating by recipient and need to be careful of requesting more than 10,000 buckets?